### PR TITLE
Fix entrypoint definition in pyproject.toml 

### PR DESCRIPTION
**Pull request by mbargull**
### Description

Just a small amendment for the equivalent changes to `recipe/meta.yaml` from
https://github.com/conda/conda-build/pull/5151/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR12 .

### Checklist - did you ...

- ~~[ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~~
     
- ~~[ ] Add / update necessary tests?~~
- ~~[ ] Add / update outdated documentation?~~

### CODE REVIEW

**Review by kenodegard — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -55,15 +55,15 @@ repository = "https://github.com/conda/conda-build"
...
-conda-index = "conda_build.cli.main_index:main"
-conda-inspect = "conda_build.cli.main_inspect:main"
-conda-metapackage = "conda_build.cli.main_metapackage:main"
-conda-render = "conda_build.cli.main_render:main"
-conda-skeleton = "conda_build.cli.main_skeleton:main"
-conda-debug = "conda_build.cli.main_debug:main"
+conda-build = "conda_build.cli.main_build:execute"
+conda-convert = "conda_build.cli.main_convert:execute"
+conda-develop = "conda_build.cli.main_develop:execute"
+conda-index = "conda_build.cli.main_index:execute"
```
Comment (mbargull) [inline pyproject.toml:61]: Oh, how do we want to go about `conda-index`?
(The same line is also still in `recipe/meta.yaml` and thus needs adjusting there too.)

The cleanest way would be to
1. add the `conda-index` script to the `conda-index` package,
2. add a `run_constrains` to `conda-index` with `conda-build>=24.1`,
3. release a new version of `conda-index` with those changes,
4. add a that version as the lower bound of the `conda-index` dependency in `conda-build`.

... But IDK, if it's feasible to require a new `conda-index` release before the impending `conda-build=24.1` release....

Comment (kenodegard) [reply pyproject.toml:61]: we are already planning on a `conda-index 0.4.0` release to resolve this (and a circular dependency issue for Python 3.12 support)

`conda-build` should no longer be making any `conda-index` entrypoint scripts

Comment (mbargull) [reply pyproject.toml:61]: If it's not possible to do a new `conda-index` release beforehand, we may want to set this as `conda-index = "conda_index.cli.cli"`?

Comment (kenodegard) [reply pyproject.toml:61]: as for it still being in `recipe/meta.yaml`, that's a mistake we need to rectify

**Review by kenodegard — APPROVED** [review]

**Pull request merged**